src/providers/wms/qgswmsprovider.cpp

```diff
@@ -1100,7 +1103,8 @@ void QgsWmsProvider::addWmstParameters( QUrlQuery &query )
     }
   }
 
-  if ( uri.param( QStringLiteral( "enableTime" ) ) == QLatin1String( "no" ) )
+  if ( uri.contains( QStringLiteral( "enableTime" ) ) &&
```

Comment:
As I mentioned in another occasion I think it's a source of potential errors and confusion the fact that we are not consistent in the usage of:
- yes/no
- true/false
- 0/1
in uris.

I would strongly recommend to be tolerant and create a function that can accept any of the above mentioned values (case-insensitive), and stick to boolean true/false when you set a value.

Note: this is absolutely not your fault, but since you are doing an amazing job here I hope you can make an extra step and pave the path for a better and consistent handling of URI parameters  (for other providers too!).

Reply:
Thanks @elpaso, I have updated the PR, now using true/false values only.

Reply:
Also @elpaso, how would want the function you have recommend to look like? something like this
```c++
void QgsWmsProviderMetadata::setBoolParam( QVariantMap uri, const QString& parameter, const QVariant& value)
{
  if ( value.toString() == QStringLiteral( "yes" ) || value.toString() == QStringLiteral( "1" ) || value.toBool() ) {
  	uri[ parameter ] = true;
  }
  else if ( value.toString() == QStringLiteral( "no" ) || value.toString() == QStringLiteral( "0" ) || !value.toBool() ) {
  	uri[ parameter ] = false;
  }
}
```

Reply:
Yes! 

but maybe better if:
1. case insensitive, you may use https://doc.qt.io/qt-5/qstring.html#compare-1 case insensitive.
2. also check for true/false (case insensitive) string values
3. rename to `setBoolParameter` (for readability)

I think we also need one for `getBoolParameter`, maybe with a default optional argument, something like:
```c++
static bool QgsProviderMetadata::boolParameter( QVariantMap uri, const QString& parameter, bool defaultValue = false )
```

Reply:
Ok @elpaso can I make these on a different PR?

Reply:
Sure, do what's more convenient to you.

Reply:
Great

Reply:
@elpaso here https://github.com/qgis/QGIS/pull/36087